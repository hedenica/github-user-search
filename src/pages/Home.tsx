import { ChangeEvent, useState } from 'react';
import { useNavigate } from 'react-router-dom';

import { Button } from '../components/Button';
import { Input } from '../components/Input';

import styles from './home.module.scss';

export const Home = () => {
  const navigate = useNavigate();
  const [username, setUserName] = useState('');

  const handleUsernameChange = (e: ChangeEvent<HTMLInputElement>) => {
    setUserName(e.target.value);
  };

  const navigateToUserPage = () => navigate(`/user/${username}`);

  return (
    <div className={styles.container}>
      <h1 className={styles.title}>Search for a user</h1>
      <div className={styles.content}>
        <Input
          customStyle={styles.input}
          type="text"
          name="username"
          placeholder="Type a github username"
          value={username}
          autoCapitalize="none"
          onChange={handleUsernameChange}
        />
        <Button
          customStyle={styles.button}
          title="search"
          isDisabled={username.length <= 3}
          onClick={navigateToUserPage}
        />
      </div>
    </div>
  );
};
