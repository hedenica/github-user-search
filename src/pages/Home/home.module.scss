.container {
  display: flex;
  width: 100%;
  height: 80vh;
  flex-direction: column;
  align-items: center;
  align-content: center;

  background-image: url('../../assets/background.png');
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
}

.title {
  color: #fff;
}

.content {
  display: flex;
}

.button {
  width: 100px;
}

@media (min-width: 320px) {
  .input {
    width: 200px;
  }
}
  
@media (min-width: 768px) {
  .input {
    width: 350px;
  }
}
  
@media (min-width: 992px) {
  .input {
    width: 400px;
  }
}
