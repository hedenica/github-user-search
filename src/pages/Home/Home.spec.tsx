import { render, screen, fireEvent } from '@testing-library/react';

import { Home } from './Home';

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockedNavigate,
  };
});

describe('Home page', () => {
  it('renders correctly', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });

  describe('When user field is filled', () => {
    describe('With more than 3 or equals to 3 characters', () => {
      it('shows button search as enabled', () => {
        render(<Home />);

        const input = screen.getByPlaceholderText(/type a github username/i);

        fireEvent.change(input, { target: { value: 'hedenica' } });

        const searchButton = screen.getByRole('button');

        expect(searchButton).toBeEnabled();
      });

      it('redirects to user page', () => {
        render(<Home />);

        const input = screen.getByPlaceholderText(/type a github username/i);

        fireEvent.change(input, { target: { value: 'hedenica' } });

        const searchButton = screen.getByRole('button');

        fireEvent.click(searchButton);

        expect(mockedNavigate).toHaveBeenCalledWith('/user/hedenica');
      });
    });

    describe('With less than 4 characters', () => {
      it('shows button search as enabled', () => {
        render(<Home />);

        const input = screen.getByPlaceholderText(/type a github username/i);

        fireEvent.change(input, { target: { value: 'hed' } });

        const searchButton = screen.getByRole('button');

        expect(searchButton).toBeDisabled();
      });
    });
  });
});
