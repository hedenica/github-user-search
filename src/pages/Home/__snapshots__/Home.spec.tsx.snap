// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<div>
  <div>
    <h1>
      Search for a user
    </h1>
    <div>
      <div>
        <input
          autocapitalize="none"
          class=""
          name="username"
          placeholder="Type a github username"
          type="text"
          value=""
        />
      </div>
      <button
        class=""
        disabled=""
        type="button"
      >
        search
      </button>
    </div>
  </div>
</div>
`;
